tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -272,8 +272,123 @@ constexpr void test_compiletime() {
     }
 }
 static_assert((test_compiletime(), true));
+
+template <class T>
+struct A {
+    T value;
+
+    constexpr A() noexcept = default;
+    constexpr ~A()         = default;
+};
+
+template <class T>
+struct nontrivial_A {
+    T value;
+
+    constexpr nontrivial_A(T in = T{}) noexcept : value(in){};
+    constexpr ~nontrivial_A() {
+        value.~T();
+    };
+};
+
+template <class T>
+struct Alloc {
+    using value_type = T;
+    using size_type  = std::size_t;
+
+    constexpr Alloc(int id_) noexcept : id(id_) {}
+    constexpr Alloc(const Alloc&) = default;
+    constexpr ~Alloc()            = default;
+
+    constexpr value_type* allocate(size_t n) {
+        assert(n == 10);
+        return static_cast<value_type*>(::operator new(n * sizeof(value_type)));
+    }
+
+    constexpr void deallocate(value_type* ptr, size_t n) {
+        assert(n == 10);
+        ::operator delete(ptr, n * sizeof(value_type));
+    }
+
+    constexpr Alloc<T> select_on_container_copy_construction() const noexcept {
+        return Alloc<T>{id + 1};
+    }
+
+    constexpr size_type max_size() noexcept {
+        return std::numeric_limits<size_type>::max() / sizeof(value_type);
+    }
+

```

Comment:
I believe we should also have test for an allocator with/without a `construct` and `destroy` methods, to ensure that they are properly forwarded.

I guess two additional bool template arguments and some SFINAE magic are the least pain here

Reply:
I added 3 more cases with/without `construct` and `destroy` methods.